src/Process.js

```diff
@@ -1,59 +1,66 @@
-const { SchedulerInterrupt } = require('./constants/index');
+const { SchedulerInterrupt } = require('./constants/index')
 
 // A class representation of a process that may be blocking
 // or non-blocking. We can specify how much CPU time a process
 // needs in order to complete, or we can specify if the process
 // is blocking; if so, the amount of blocking time needed is
 // randomly determined.
 class Process {
-    constructor(pid, cpuTimeNeeded=null, blocking=false) {
-        this._pid = pid;
-        this.queue = null;
-        this.cpuTimeNeeded = (cpuTimeNeeded !== null) ? cpuTimeNeeded : Math.round(Math.random() * 1000);
-        this.blockingTimeNeeded = blocking ? Math.round(Math.random() * 100) : 0;
-        // A bool representing whether this process was toggled from blocking to non-blocking or vice versa
-        this.stateChanged = false;
-    }
-    
-    setParentQueue(queue) {
-
-    }
+  constructor(pid, cpuTimeNeeded = null, blocking = false) {
+    this._pid = pid
+    this.queue = null
+    this.cpuTimeNeeded =
+      cpuTimeNeeded !== null ? cpuTimeNeeded : Math.round(Math.random() * 1000)
+    this.blockingTimeNeeded = blocking ? Math.round(Math.random() * 100) : 0
+    // A bool representing whether this process was toggled from blocking to non-blocking or vice versa
+    this.stateChanged = false
+  }
 
-    isFinished() {
+  setParentQueue(queue) {
+    this.queue = queue
+  }
 
+  isFinished() {
+    if (!this.cpuTimeNeeded && !this.blockingTimeNeeded && !this.queue) {
```

Comment:
This is a very readable version of the logic, I like it. You could make it even more succinct than if if/else statement, though...